src/AzurePipelines.ml

```diff
@@ -0,0 +1,118 @@
+open Bindings
+
+module E = struct
+  type t =
+    | DownloadFailure of string
+    | UnsupportedOS
+    | InvalidJSONType of string
+    | MissingField of string
+    | InvalidFirstArrayElement
+    | Failure of string
+
+  let toString = function
+    | DownloadFailure url -> {j| Could not download $url |j}
+    | UnsupportedOS ->
+      "We detected a platform for which we couldn't find cached builds"
+    | InvalidJSONType value ->
+      {j|Field $value in Azure's response was undefined|j}
+    | MissingField k -> {j| Response from Azure did not contain build $k |j}
+    | InvalidFirstArrayElement -> "Unexpected array value in Azure response"
+    | Failure url -> {j| Failed to download $url |j}
+end
+
+module JSONResponse = struct
+  module CamlArray = Array
+  open Js
+
+  let getBuildId responseText =
```

Comment:
Do you think this function is easy to understand?

Reply:
Ported the JSON parsing logic to bs-json. This should make it easier to read.